pkg/url/url.go

```diff
@@ -216,25 +337,42 @@ func ConvertConfigURL(configURL config.ConfigURL) URL {
 }
 
 // GetURLString returns a string representation of given url
-func GetURLString(protocol, URL string) string {
+func GetURLString(protocol, URL string, ingressDomain string) string {
+	if experimental.IsExperimentalModeEnabled() {
+		return protocol + "://" + ingressDomain
+	}
 	return protocol + "://" + URL
 }
 
 // Exists checks if the url exists in the component or not
 // urlName is the name of the url for checking
 // componentName is the name of the component to which the url's existence is checked
 // applicationName is the name of the application to which the url's existence is checked
-func Exists(client *occlient.Client, urlName string, componentName string, applicationName string) (bool, error) {
-	urls, err := ListPushed(client, componentName, applicationName)
-	if err != nil {
-		return false, errors.Wrap(err, "unable to list the urls")
-	}
+func Exists(client *occlient.Client, kClient *kclient.Client, urlName string, componentName string, applicationName string) (bool, error) {
```

Comment:
Please refactor the function and use the get functions instead of the List ones.